File: java/src/test/java/com/katas/supermarket/api/ChangeMeControllerIntegrationTest.java
package com.katas.supermarket.api;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.test.web.servlet.MockMvc;
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;

import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

@WebMvcTest(ChangeMeController.class)
public class ChangeMeControllerIntegrationTest {

    @Autowired
    private MockMvc mockMvc;

    @Test
    public void should_not_calculate_anything_if_cart_is_empty() throws Exception {
        mockMvc.perform(
                MockMvcRequestBuilders.post("/checkout").content("{ \"skus\": [] }")
        ).andExpect(status().isNoContent());
    }

    // TODO: Start here adding next test
}

File: kotlin/src/test/kotlin/com/katas/supermarket/api/ChangeMeControllerIntegrationTest.kt
package com.katas.supermarket.api

import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post
import org.springframework.test.web.servlet.result.MockMvcResultMatchers.status

@WebMvcTest(ChangeMeController::class)
class ChangeMeControllerIntegrationTest {

    @Autowired
    private lateinit var mockMvc: MockMvc

    @Test
    fun `should not calculate anything if the cart is empty of products`() {
        val result = mockMvc.perform(
                post("/checkout").content("""{ "skus": [] }""")
        )

        result.andExpect(status().isNoContent)
    }
}
